cdapython/results/result.py

```diff
@@ -1,83 +1,65 @@
 """
-result is a convenient wrapper around the response object from the CDA service it adds user functionality.
-like creating dataframe and manipulating data for ease-of-use such as paginating automatically for the user through their result objects.
+result is a convenient wrapper around the response object
+from the CDA service it adds user functionality.
+like creating dataframe and manipulating data for ease-of-use such
+as paginating automatically for the user through their result objects.
 """
-import json
-import re
+from __future__ import annotations
+
 from collections import ChainMap
 from io import StringIO
 from multiprocessing.pool import ApplyResult
 from time import sleep
-from typing import (
-    Any,
-    AsyncGenerator,
-    Dict,
-    Iterator,
-    List,
-    Optional,
-    Pattern,
-    Type,
-    Union,
-)
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Type, Union
 
 from cda_client.api.query_api import QueryApi
 from cda_client.model.query_response_data import QueryResponseData
-from pandas import DataFrame, MultiIndex, Series, json_normalize, read_csv
-from rich import print
-from rich.table import Table
+from pandas import DataFrame, read_csv
 from typing_extensions import Literal
 
 from cdapython.Paginator import Paginator
+from cdapython.results import COLLECT_RESULT
 from cdapython.results.base import BaseResult
-from cdapython.utils.state import State
-
+from cdapython.results.factories.result_factory import ResultFactory
 
-class _QEncoder(json.JSONEncoder):
-    """_QEncoder this is a a class to help with json conversion
-        the standard json dump
-
-    Args:
-        json (_type_): _description_
-    """
-
-    def decode(self, object_json) -> str:
-        """
-        This is for json decoding super method
-        """
-        regex: Pattern[str] = re.compile(r"([^\"\\]|\\([\"\\\/bfnrt]|u[a-zA-Z\d]{4}))+")
-
-        if isinstance(object_json, str):
-            print(regex.match(object_json, pos=0))
-        return object_json
+if TYPE_CHECKING:
+    from cdapython.results.columns_result import ColumnsResult
+    from cdapython.results.factories.collect_result import CollectResult
+    from cdapython.results.string_result import StringResult
 
 
 class Result(BaseResult):
     """
-    The Results Class is a convenient wrapper around the response object from the CDA service.
+    The Results Class is a convenient wrapper around the
+    response object from the CDA service.
     """
 
     def __init__(
         self,
         api_response: QueryResponseData,
         query_id: str,
-        offset: Optional[int],
-        limit: Optional[int],
+        offset: int,
+        limit: int,
```

Comment:
why are these no longer optional? how we will know whether a user wants a specific value or not? I assume the default for each is 0.